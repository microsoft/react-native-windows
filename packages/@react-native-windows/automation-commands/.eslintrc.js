module.exports = {
    extends: ['@rnw-scripts'],
    parserOptions: {tsconfigRootDir : __dirname},
};
