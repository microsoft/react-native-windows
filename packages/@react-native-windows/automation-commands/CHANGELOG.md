# Change Log - @react-native-windows/automation-commands

This log was last generated on Fri, 01 Oct 2021 05:05:54 GMT and should not be manually modified.

<!-- Start content -->

## 0.0.6-2

Fri, 01 Oct 2021 05:05:54 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.5-2 (julio.rocha@microsoft.com)

## 0.0.6-1

Thu, 30 Sep 2021 05:06:48 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.5-1 (erozell@outlook.com)

## 0.0.6-0

Wed, 29 Sep 2021 05:06:54 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.5-0 (anandrag@microsoft.com)

## 0.0.5

Tue, 28 Sep 2021 05:07:03 GMT

### Patches

- Bump @react-native-windows/automation-channel to v0.0.4 (34109996+chiaramooney@users.noreply.github.com)

## 0.0.5-0

Sat, 25 Sep 2021 05:06:39 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-12 (53799235+ZihanChen-MSFT@users.noreply.github.com)

## 0.0.4

Fri, 24 Sep 2021 05:06:42 GMT

### Patches

- Bump @react-native-windows/automation-channel to v0.0.3-11 (igklemen@microsoft.com)

## 0.0.4-1

Thu, 23 Sep 2021 05:06:48 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-10 (34109996+chiaramooney@users.noreply.github.com)

## 0.0.4-0

Wed, 22 Sep 2021 05:06:43 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-9 (ali-hk@users.noreply.github.com)

## 0.0.3

Tue, 21 Sep 2021 05:08:39 GMT

### Patches

- Bump @react-native-windows/automation-channel to v0.0.3-8 (igklemen@microsoft.com)

## 0.0.3-7

Sat, 18 Sep 2021 05:07:45 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-7 (erozell@outlook.com)

## 0.0.3-6

Fri, 17 Sep 2021 05:06:53 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-6 (jthysell@microsoft.com)

## 0.0.3-5

Thu, 16 Sep 2021 05:06:44 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-5 (erozell@outlook.com)

## 0.0.3-4

Wed, 15 Sep 2021 05:07:04 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-4 (30809111+acoates-ms@users.noreply.github.com)

## 0.0.3-3

Tue, 14 Sep 2021 05:07:02 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-3 (jthysell@microsoft.com)

## 0.0.3-2

Sat, 11 Sep 2021 05:06:55 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-2 (anandrag@microsoft.com)

## 0.0.3-1

Fri, 10 Sep 2021 05:07:31 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-1 (53799235+ZihanChen-MSFT@users.noreply.github.com)

## 0.0.3-0

Thu, 09 Sep 2021 05:07:46 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-0 (53799235+ZihanChen-MSFT@users.noreply.github.com)

## 0.0.2

Wed, 08 Sep 2021 05:08:53 GMT

### Patches

- Set consistent node requirements on our packages (ngerlem@microsoft.com)

### Changes

- Bump @react-native-windows/automation-channel to v0.0.2 (ngerlem@microsoft.com)

## 0.0.2-0

Sat, 04 Sep 2021 05:06:39 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.2-0 (dannyvv@microsoft.com)

## 0.0.1

Fri, 03 Sep 2021 05:06:04 GMT

### Patches

- @react-native-windows/automation (ngerlem@microsoft.com)

### Changes

- Bump @react-native-windows/automation-channel to v0.0.1 (erozell@outlook.com)
