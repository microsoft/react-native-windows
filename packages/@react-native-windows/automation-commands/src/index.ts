/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import dumpVisualTree from './dumpVisualTree';

export {dumpVisualTree};
