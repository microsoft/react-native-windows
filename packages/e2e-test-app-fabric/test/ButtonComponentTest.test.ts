/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import {dumpVisualTree} from '@react-native-windows/automation-commands';
import {goToComponentExample} from './RNTesterNavigation';
import {verifyNoErrorLogs} from './Helpers';
import {app} from '@react-native-windows/automation';

beforeAll(async () => {
  // If window is partially offscreen, tests will fail to click on certain elements
  await app.setWindowPosition(0, 0);
  await app.setWindowSize(1000, 1250);
  await goToComponentExample('Button');
});

afterEach(async () => {
  await verifyNoErrorLogs();
});

describe('Button Tests', () => {
  test('Buttons have default styling', async () => {
    const component = await app.findElementByTestID('button_default_styling');
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('button_default_styling');
    expect(dump).toMatchSnapshot();
  });
  test('Buttons can have custom colors', async () => {
    const component = await app.findElementByTestID('cancel_button');
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('cancel_button');
    expect(dump).toMatchSnapshot();
  });
  test('Buttons can be disabled', async () => {
    const component = await app.findElementByTestID('disabled_button');
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('disabled_button');
    expect(dump).toMatchSnapshot();
  });
  test('Buttons can have accessibility labels', async () => {
    const component = await app.findElementByTestID(
      'accessibilityLabel_button',
    );
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('accessibilityLabel_button');
    expect(dump).toMatchSnapshot();
  });
  test('Buttons can have accessibility states', async () => {
    const component = await app.findElementByTestID(
      'accessibilityState_button',
    );
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('accessibilityState_button');
    expect(dump).toMatchSnapshot();
  });
  test('Buttons can have accessibility props', async () => {
    const component = await app.findElementByTestID('accessibility_props');
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('accessibility_props');
    expect(dump).toMatchSnapshot();
  });
});
