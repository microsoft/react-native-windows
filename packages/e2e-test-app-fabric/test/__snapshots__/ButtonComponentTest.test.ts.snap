// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Buttons can be disabled 1`] = `
{
  "Automation Tree": {
    "AutomationId": "disabled_button",
    "Children": [
      {
        "AutomationId": "",
        "ControlType": 50020,
        "HelpText": "",
        "IsEnabled": false,
        "IsKeyboardFocusable": false,
        "LocalizedControlType": "text",
        "Name": "Submit Application",
      },
    ],
    "ControlType": 50000,
    "HelpText": "",
    "IsEnabled": false,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
    "Name": "Press to submit your application!",
  },
  "Visual Tree": {
    "Children": [
      {
        "Children": [
          {
            "Children": [
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  1,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -1,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  -2,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  2,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
            ],
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              916,
              37,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          0,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          916,
          37,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Children": [
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              898,
              20,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          9,
          9,
          0,
        ],
        "Opacity": 1,
        "Size": [
          898,
          20,
        ],
        "Visual Type": "SpriteVisual",
      },
    ],
    "Comment": "disabled_button",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      916,
      37,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`Button Tests Buttons can have accessibility labels 1`] = `
{
  "Automation Tree": {
    "AutomationId": "accessibilityLabel_button",
    "Children": [
      {
        "AutomationId": "",
        "ControlType": 50020,
        "HelpText": "",
        "IsEnabled": true,
        "IsKeyboardFocusable": false,
        "LocalizedControlType": "text",
        "Name": "Submit Application",
      },
    ],
    "ControlType": 50000,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
    "Name": "Press to submit your application!",
  },
  "Visual Tree": {
    "Children": [
      {
        "Children": [
          {
            "Brush": {
              "Brush Type": "ColorBrush",
              "Color": "rgba(0, 122, 255, 255)",
            },
            "Children": [
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  1,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -1,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  -2,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  2,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
            ],
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              916,
              37,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          0,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          916,
          37,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Children": [
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              898,
              20,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          9,
          9,
          0,
        ],
        "Opacity": 1,
        "Size": [
          898,
          20,
        ],
        "Visual Type": "SpriteVisual",
      },
    ],
    "Comment": "accessibilityLabel_button",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      916,
      37,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`Button Tests Buttons can have accessibility props 1`] = `
{
  "Automation Tree": {
    "AutomationId": "accessibility_props",
    "Children": [
      {
        "AutomationId": "",
        "ControlType": 50020,
        "HelpText": "",
        "IsEnabled": true,
        "IsKeyboardFocusable": false,
        "LocalizedControlType": "text",
        "Name": "Submit Application",
      },
    ],
    "ControlType": 50000,
    "HelpText": "Submit Button",
    "IsEnabled": true,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
    "Name": "Press to submit your application!",
  },
  "Visual Tree": {
    "Children": [
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          0,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          4,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          4,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          908,
          1,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          -4,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          4,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          -1,
          4,
          0,
        ],
        "Opacity": 1,
        "Size": [
          1,
          28,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          -4,
          -5,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          5,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          4,
          -2,
          0,
        ],
        "Opacity": 1,
        "Size": [
          908,
          2,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          0,
          -5,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          5,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          0,
          4,
          0,
        ],
        "Opacity": 1,
        "Size": [
          1,
          28,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Children": [
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              898,
              20,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          9,
          9,
          0,
        ],
        "Opacity": 1,
        "Size": [
          898,
          20,
        ],
        "Visual Type": "SpriteVisual",
      },
    ],
    "Comment": "accessibility_props",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      916,
      37,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`Button Tests Buttons can have accessibility states 1`] = `
{
  "Automation Tree": {
    "AutomationId": "accessibilityState_button",
    "Children": [
      {
        "AutomationId": "",
        "ControlType": 50020,
        "HelpText": "",
        "IsEnabled": false,
        "IsKeyboardFocusable": false,
        "LocalizedControlType": "text",
        "Name": "Submit Application",
      },
    ],
    "ControlType": 50000,
    "HelpText": "",
    "IsEnabled": false,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
    "Name": "Press to submit your application!",
  },
  "Visual Tree": {
    "Children": [
      {
        "Children": [
          {
            "Children": [
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  1,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -1,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  -2,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  2,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
            ],
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              916,
              37,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          0,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          916,
          37,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Children": [
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              898,
              20,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          9,
          9,
          0,
        ],
        "Opacity": 1,
        "Size": [
          898,
          20,
        ],
        "Visual Type": "SpriteVisual",
      },
    ],
    "Comment": "accessibilityState_button",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      916,
      37,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`Button Tests Buttons can have custom colors 1`] = `
{
  "Automation Tree": {
    "AutomationId": "cancel_button",
    "Children": [
      {
        "AutomationId": "",
        "ControlType": 50020,
        "HelpText": "",
        "IsEnabled": true,
        "IsKeyboardFocusable": false,
        "LocalizedControlType": "text",
        "Name": "Cancel Application",
      },
    ],
    "ControlType": 50000,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
    "Name": "Press to cancel your application!",
  },
  "Visual Tree": {
    "Children": [
      {
        "Children": [
          {
            "Brush": {
              "Brush Type": "ColorBrush",
              "Color": "rgba(255, 59, 48, 255)",
            },
            "Children": [
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  1,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  0,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  4,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -1,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  -4,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  4,
                  -2,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  908,
                  2,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  -5,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  4,
                  5,
                ],
                "Visual Type": "SpriteVisual",
              },
              {
                "Brush": {
                  "Brush Type": "ColorBrush",
                  "Color": "rgba(0, 0, 0, 255)",
                },
                "Offset": [
                  0,
                  4,
                  0,
                ],
                "Opacity": 1,
                "Size": [
                  1,
                  28,
                ],
                "Visual Type": "SpriteVisual",
              },
            ],
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              916,
              37,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          0,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          916,
          37,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Children": [
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              898,
              20,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          9,
          9,
          0,
        ],
        "Opacity": 1,
        "Size": [
          898,
          20,
        ],
        "Visual Type": "SpriteVisual",
      },
    ],
    "Comment": "cancel_button",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      916,
      37,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`Button Tests Buttons have default styling 1`] = `
{
  "Automation Tree": {
    "AutomationId": "button_default_styling",
    "Children": [
      {
        "AutomationId": "",
        "ControlType": 50020,
        "HelpText": "",
        "IsEnabled": true,
        "IsKeyboardFocusable": false,
        "LocalizedControlType": "text",
        "Name": "Submit Application",
      },
    ],
    "ControlType": 50000,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
    "Name": "Press to submit your application!",
  },
  "Visual Tree": {
    "Children": [
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          0,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          4,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          4,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          908,
          1,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          -4,
          0,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          4,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          -1,
          4,
          0,
        ],
        "Opacity": 1,
        "Size": [
          1,
          28,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          -4,
          -5,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          5,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          4,
          -2,
          0,
        ],
        "Opacity": 1,
        "Size": [
          908,
          2,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          0,
          -5,
          0,
        ],
        "Opacity": 1,
        "Size": [
          4,
          5,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Brush": {
          "Brush Type": "ColorBrush",
          "Color": "rgba(0, 0, 0, 255)",
        },
        "Offset": [
          0,
          4,
          0,
        ],
        "Opacity": 1,
        "Size": [
          1,
          28,
        ],
        "Visual Type": "SpriteVisual",
      },
      {
        "Children": [
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              898,
              20,
            ],
            "Visual Type": "SpriteVisual",
          },
          {
            "Offset": [
              0,
              0,
              0,
            ],
            "Opacity": 1,
            "Size": [
              0,
              0,
            ],
            "Visual Type": "SpriteVisual",
          },
        ],
        "Offset": [
          9,
          9,
          0,
        ],
        "Opacity": 1,
        "Size": [
          898,
          20,
        ],
        "Visual Type": "SpriteVisual",
      },
    ],
    "Comment": "button_default_styling",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      916,
      37,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;
